jina/drivers/evaluate.py

```diff
@@ -0,0 +1,22 @@
+__copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+from typing import Iterable, Tuple
+from jina.proto import jina_pb2
+
+from . import BaseExecutableDriver
+
+
+class EvaluateDriver(BaseExecutableDriver):
+    """Drivers inherited from this Driver will bind :meth:`evaluate` by default
+    """
+
+    def __init__(self, executor: str = None, traversal_paths: Tuple[str] = ('r',), method: str = 'evaluate', *args,
+                 **kwargs):
+        super().__init__(executor, method, traversal_paths=traversal_paths, *args, **kwargs)
+        self._is_apply = True
```

Comment:
`False`?

Reply:
> I think we should differentiate between evaluation as suggested here and tracking live KPIs (e.g. for an A/B test or regular monitoring) which would be rather fire-and-forget messages to pods and therefor would need another ending as you suggested in _Future work_.
> 
> The in-memory count must not only be averages, but also the concrete scores. You cannot compute some KPIs with just averages (See NDCG).

When I mean average, is to aggregate all NDCG of all queries to have a unique value for the system. It can be flexible but I do not see where it is not applicable that u want the average on all queries